ToolBox/TickAggregatingConsolidator.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.ToolBox
+{
+    /// <summary>
+    /// Class that uses consolidators to aggregate tick data data
+    /// </summary>
+    public abstract class TickAggregatingConsolidator
+    {
+        /// <summary>
+        /// The consolidator used to aggregate data from
+        /// higher resolutions to data in lower resolutions
+        /// </summary>
+        public abstract IDataConsolidator Consolidator { get; }
```

Comment:
Consider setting this via a constructor argument in this class instead of having it abstract